pytroll_collectors/tests/test_segments.py

```diff
@@ -988,13 +1016,27 @@ def test_mismatching_files_generate_multiple_slots(self):
         """Test mismatching files generate multiple slots."""
         from posttroll.message import Message as Message_p
         viirs_msg = Message_p(rawstr=viirs_message)
-        pps_msg = FakeMessage(pps_message_data, message_type='dataset')
+        pps_msg = FakeMessage(pps_message_data, message_type='dataset', subject='/foo/pps')
 
         self.collection_gatherer.process(viirs_msg)
         self.collection_gatherer.process(pps_msg)
 
         assert len(self.collection_gatherer.slots) == 2
 
+    def test_mismatching_files_within_tolerance_generate_one_slot(self):
+        """Test mismatching files within timetolerance generate one slot."""
+        from posttroll.message import Message as Message_p
+        viirs_msg = Message_p(rawstr=viirs_message)
+        pps_msg = FakeMessage(new_pps_message_data_within_time_tolerance, message_type='dataset', subject='/foo/pps')
+
+        self.collection_gatherer.process(viirs_msg)
+        self.collection_gatherer.process(pps_msg)
+
+        assert len(self.collection_gatherer.slots) == 1
+        slot = self.collection_gatherer.slots['2020-10-13 05:17:21.200000']
+        assert slot.output_metadata['collection']['pps']['dataset'] != []
+        # assert slot.get_status() == Status.SLOT_READY
```

Comment:
What about this comment? Something you had in mind implementing?
And the last check only tests that the variable is not an empty list. Can this be more explicit?

Reply:
I fixed the time tolerance half-way. The status is not set to ready even though the correct files are found. Adding the  assert slot.get_status() == Status.SLOT_READY will make the test fail as step 1 in TDD. But to fix it I think is more difficult. So we decided to make an issue instead https://github.com/pytroll/pytroll-collectors/issues/129 so that this PR could be merged. I left the line in the code for later.